components/tools/OmeroPy/src/omero/plugins/upload.py

```diff
@@ -64,9 +70,27 @@ def upload(self, args):
                              " import")
             else:
                 obj = client.upload(file, type=omero_format)
-                self.ctx.out("Uploaded %s as " % file + str(obj.id.val))
+                fileIds.append(obj.id.val)
                 self.ctx.set("last.upload.id", obj.id.val)
 
+        if args.annotate:
+            c = self.ctx.conn(args)
+            update = c.sf.getUpdateService()
+            query = c.sf.getQueryService()
+
+            for fId in fileIds:
+                # obj new FileAnnotation file=OriginalFile:275
+                ann = FileAnnotationI()
+                # f = file get by id
+                ann.file = query.find("OriginalFile", fId)
```

Comment:
Yes, that form works fine, I'd just reused a pattern from elsewhere without looking any further. I've added a commit to use this neater pattern. And removed a couple of my temporary note comments.